server/hotrod/src/main/scala/org/infinispan/server/hotrod/ClientListenerRegistry.scala

```diff
@@ -195,9 +195,12 @@ class ClientListenerRegistry(configuration: HotRodServerConfiguration) extends L
    }
 
    def findAndWriteEvents(channel: Channel): Unit = {
-      eventSenders.values().collectFirst { case s: BaseClientEventSender =>
-         if(s.hasChannel(channel)) s.writeEventsIfPossible()
-      }
+      // Make sure we write any event in main event loop
+      channel.eventLoop().execute(new Runnable {
+         override def run(): Unit = eventSenders.values().collectFirst {
+            case s: BaseClientEventSender => if (s.hasChannel(channel)) s.writeEventsIfPossible()
+         }
+      })
```

Comment:
@galderz I think that with this change the ClientStressTest would pass on master, wouldn't it?